### PR TITLE
Added more description on how to get slugs

### PR DESCRIPTION
**Pull request by jjasghar**
Finding the slugs wasn't delightful, this might help someone in the future.

Signed-off-by: JJ Asghar <jj@chef.io>

### CODE REVIEW

**Comment by t0mk** [comment]
this is part of a bigger problem of visibility of what OS is deployable on what plan (plus the slugs as you mentioned). At the moment, it's:

```[{'distro': 'alpine',
  'name': 'Alpine 3',
  'provisionable_on': [],
  'slug': 'alpine_3',
  'version': '3'},
 {'distro': 'centos',
  'name': 'CentOS 6',
  'provisionable_on': [],
  'slug': 'centos_6',
  'version': '6'},
 {'distro': 'centos',
  'name': 'CentOS 7',
  'provisionable_on': ['c1.large.arm',
                       'baremetal_2a',
                       'c1.small.x86',
                       'baremetal_1',
                       'c1.xlarge.x86',
                       'baremetal_3',
                       'c2.medium.x86',
                       'd1f.optane.x86',
                       'd1p.optane.x86',
                       'm1.xlarge.x86',
                       'baremetal_2',
                       'm2.xlarge.x86',
                       's1.large.x86',
                       'baremetal_s',
                       't1.small.x86',
                       'baremetal_0',
                       'x1.small.x86',
                       'baremetal_1e'],
  'slug': 'centos_7',
  'version': '7'},
 {'distro': 'coreos',
  'name': 'Container Linux - Alpha',
  'provisionable_on': ['c1.small.x86',
                       'baremetal_1',
                       'c1.xlarge.x86',
                       'baremetal_3',
                       'c2.medium.x86',
                       'm1.xlarge.x86',
                       'baremetal_2',
                       'm2.xlarge.x86',
                       's1.large.x86',
                       'baremetal_s',
                       't1.small.x86',
                       'baremetal_0',
                       'x1.small.x86',
                       'baremetal_1e'],
  'slug': 'coreos_alpha',
  'version': 'alpha'},
 {'distro': 'coreos',
  'name': 'Container Linux - Beta',
  'provisionable_on': ['c1.small.x86',
                       'baremetal_1',
                       'c1.xlarge.x86',
                       'baremetal_3',
                       'c2.medium.x86',
                       'm1.xlarge.x86',
                       'baremetal_2',
                       'm2.xlarge.x86',
                       's1.large.x86',
                       'baremetal_s',
                       't1.small.x86',
                       'baremetal_0'],
  'slug': 'coreos_beta',
  'version': 'beta'},
 {'distro': 'coreos',
  'name': 'Container Linux - Stable',
  'provisionable_on': ['c1.small.x86',
                       'baremetal_1',
                       'c1.xlarge.x86',
                       'baremetal_3',
                       'c2.medium.x86',
                       'm1.xlarge.x86',
                       'baremetal_2',
                       'm2.xlarge.x86',
                       't1.small.x86',
                       'baremetal_0'],
  'slug': 'coreos_stable',
  'version': 'stable'},
 {'distro': 'custom_ipxe',
  'name': 'Custom iPXE',
  'provisionable_on': ['baremetal_2a4',
                       'baremetal_2a5',
                       'baremetal_5',
                       'c1.large.arm',
                       'baremetal_2a',
                       'c1.large.arm.xda',
                       'baremetal_2a2',
                       'c1.small.x86',
                       'baremetal_1',
                       'c1.xlarge.x86',
                       'baremetal_3',
                       'c2.medium.x86',
                       'd1f.optane.x86',
                       'd1p.optane.x86',
                       'm1.xlarge.x86',
                       'baremetal_2',
                       'm2.xlarge.x86',
                       's1.large.x86',
                       'baremetal_s',
                       't1.small.x86',
                       'baremetal_0',
                       'x1.small.x86',
                       'baremetal_1e'],
  'slug': 'custom_ipxe',
  'version': '1'},
 {'distro': 'debian',
  'name': 'Debian 8',
  'provisionable_on': ['c1.large.arm',
                       'baremetal_2a',
                       'c1.small.x86',
                       'baremetal_1',
                       'c1.xlarge.x86',
                       'baremetal_3',
                       'c2.medium.x86',
                       'm1.xlarge.x86',
                       'baremetal_2',
                       'm2.xlarge.x86',
                       't1.small.x86',
                       'baremetal_0',
                       'x1.small.x86',
                       'baremetal_1e'],
  'slug': 'debian_8',
  'version': '8'},
 {'distro': 'debian',
  'name': 'Debian 9',
  'provisionable_on': ['c1.large.arm',
                       'baremetal_2a',
                       'c1.small.x86',
                       'baremetal_1',
                       'c1.xlarge.x86',
                       'baremetal_3',
                       'c2.medium.x86',
                       'm1.xlarge.x86',
                       'baremetal_2',
                       'm2.xlarge.x86',
                       's1.large.x86',
                       'baremetal_s',
                       't1.small.x86',
                       'baremetal_0',
                       'x1.small.x86',
                       'baremetal_1e'],
  'slug': 'debian_9',
  'version': '9'},
 {'distro': 'centos',
  'name': 'Deprovision',
  'provisionable_on': [],
  'slug': 'deprovision',
  'version': ''},
 {'distro': 'freebsd',
  'name': 'FreeBSD 10.3',
  'provisionable_on': ['c1.small.x86',
                       'baremetal_1',
                       'c1.xlarge.x86',
                       'baremetal_3',
                       'm1.xlarge.x86',
                       'baremetal_2',
                       'm2.xlarge.x86',
                       's1.large.x86',
                       'baremetal_s',
                       't1.small.x86',
                       'baremetal_0'],
  'slug': 'freebsd_10_3',
  'version': '10.3'},
 {'distro': 'freebsd',
  'name': 'FreeBSD 10.4',
  'provisionable_on': ['c1.small.x86',
                       'baremetal_1',
                       'c1.xlarge.x86',
                       'baremetal_3',
                       'm1.xlarge.x86',
                       'baremetal_2',
                       'm2.xlarge.x86',
                       's1.large.x86',
                       'baremetal_s',
                       't1.small.x86',
                       'baremetal_0'],
  'slug': 'freebsd_10_4',
  'version': '10.4'},
 {'distro': 'freebsd',
  'name': 'FreeBSD 11.0',
  'provisionable_on': ['c1.small.x86',
                       'baremetal_1',
                       'c1.xlarge.x86',
                       'baremetal_3',
                       'm1.xlarge.x86',
                       'baremetal_2',
                       'm2.xlarge.x86',
                       's1.large.x86',
                       'baremetal_s',
                       't1.small.x86',
                       'baremetal_0'],
  'slug': 'freebsd_11_0',
  'version': '11.0'},
 {'distro': 'freebsd',
  'name': 'FreeBSD 11.1',
  'provisionable_on': ['c1.small.x86',
                       'baremetal_1',
                       'c1.xlarge.x86',
                       'baremetal_3',
                       'm1.xlarge.x86',
                       'baremetal_2',
                       'm2.xlarge.x86',
                       's1.large.x86',
                       'baremetal_s',
                       't1.small.x86',
                       'baremetal_0'],
  'slug': 'freebsd_11_1',
  'version': '11.1'},
 {'distro': 'freebsd',
  'name': 'FreeBSD 12-testing',
  'provisionable_on': ['c1.large.arm', 'baremetal_2a'],
  'slug': 'freebsd_12_testing',
  'version': '12-testing'},
 {'distro': 'nixos',
  'name': 'NixOS 17.03',
  'provisionable_on': ['c1.small.x86',
                       'baremetal_1',
                       'c1.xlarge.x86',
                       'baremetal_3',
                       'm1.xlarge.x86',
                       'baremetal_2',
                       't1.small.x86',
                       'baremetal_0'],
  'slug': 'nixos_17_03',
  'version': '17.03'},
 {'distro': 'nixos',
  'name': 'NixOS 18.03',
  'provisionable_on': ['c1.large.arm',
                       'baremetal_2a',
                       'c1.small.x86',
                       'baremetal_1',
                       'c1.xlarge.x86',
                       'baremetal_3',
                       'c2.medium.x86',
                       'm1.xlarge.x86',
                       'baremetal_2',
                       'm2.xlarge.x86',
                       's1.large.x86',
                       'baremetal_s',
                       't1.small.x86',
                       'baremetal_0',
                       'x1.small.x86',
                       'baremetal_1e'],
  'slug': 'nixos_18_03',
  'version': '18.03'},
 {'distro': 'opensuse',
  'name': 'OpenSUSE 42.3',
  'provisionable_on': ['c1.large.arm',
                       'baremetal_2a',
                       'c1.small.x86',
                       'baremetal_1',
                       'c1.xlarge.x86',
                       'baremetal_3',
                       'm1.xlarge.x86',
                       'baremetal_2',
                       's1.large.x86',
                       'baremetal_s',
                       't1.small.x86',
                       'baremetal_0',
                       'x1.small.x86',
                       'baremetal_1e'],
  'slug': 'opensuse_42_3',
  'version': '42.3'},
 {'distro': 'rancher',
  'name': 'RancherOS',
  'provisionable_on': ['c1.small.x86',
                       'baremetal_1',
                       'c1.xlarge.x86',
                       'baremetal_3',
                       'm1.xlarge.x86',
                       'baremetal_2',
                       'm2.xlarge.x86',
                       's1.large.x86',
                       'baremetal_s',
                       't1.small.x86',
                       'baremetal_0'],
  'slug': 'rancher',
  'version': 'latest'},
 {'distro': 'scientific',
  'name': 'Scientific Linux 6',
  'provisionable_on': ['c1.small.x86',
                       'baremetal_1',
                       'c1.xlarge.x86',
                       'baremetal_3',
                       'c2.medium.x86',
                       'm1.xlarge.x86',
                       'baremetal_2',
                       'm2.xlarge.x86',
                       't1.small.x86',
                       'baremetal_0',
                       'x1.small.x86',
                       'baremetal_1e'],
  'slug': 'scientific_6',
  'version': '6'},
 {'distro': 'ubuntu',
  'name': 'Ubuntu 14.04 LTS',
  'provisionable_on': ['baremetal_hua',
                       'c1.small.x86',
                       'baremetal_1',
                       'c1.xlarge.x86',
                       'baremetal_3',
                       'c2.medium.x86',
                       'm1.xlarge.x86',
                       'baremetal_2',
                       'm2.xlarge.x86',
                       's1.large.x86',
                       'baremetal_s',
                       't1.small.x86',
                       'baremetal_0',
                       'x1.small.x86',
                       'baremetal_1e'],
  'slug': 'ubuntu_14_04',
  'version': '14.04'},
 {'distro': 'ubuntu',
  'name': 'Ubuntu 16.04 LTS',
  'provisionable_on': ['baremetal_2a4',
                       'baremetal_2a5',
                       'baremetal_5',
                       'baremetal_hua',
                       'c1.large.arm',
                       'baremetal_2a',
                       'c1.large.arm.xda',
                       'baremetal_2a2',
                       'c1.small.x86',
                       'baremetal_1',
                       'c1.xlarge.x86',
                       'baremetal_3',
                       'c2.medium.x86',
                       'd1f.optane.x86',
                       'd1p.optane.x86',
                       'm1.xlarge.x86',
                       'baremetal_2',
                       'm2.xlarge.x86',
                       's1.large.x86',
                       'baremetal_s',
                       't1.small.x86',
                       'baremetal_0',
                       'x1.small.x86',
                       'baremetal_1e'],
  'slug': 'ubuntu_16_04',
  'version': '16.04'},
 {'distro': 'ubuntu',
  'name': 'Ubuntu 17.10',
  'provisionable_on': ['baremetal_5',
                       'baremetal_hua',
                       'c1.small.x86',
                       'baremetal_1',
                       'c1.xlarge.x86',
                       'baremetal_3',
                       'c2.medium.x86',
                       'm1.xlarge.x86',
                       'baremetal_2',
                       'm2.xlarge.x86',
                       's1.large.x86',
                       'baremetal_s',
                       't1.small.x86',
                       'baremetal_0',
                       'x1.small.x86',
                       'baremetal_1e'],
  'slug': 'ubuntu_17_10',
  'version': '17.10'},
 {'distro': 'ubuntu',
  'name': 'Ubuntu 18.04 LTS',
  'provisionable_on': ['c1.large.arm',
                       'baremetal_2a',
                       'c1.small.x86',
                       'baremetal_1',
                       'c1.xlarge.x86',
                       'baremetal_3',
                       'c2.medium.x86',
                       'm1.xlarge.x86',
                       'baremetal_2',
                       's1.large.x86',
                       'baremetal_s',
                       't1.small.x86',
                       'baremetal_0'],
  'slug': 'ubuntu_18_04',
  'version': '18.04'},
 {'distro': 'vmware',
  'name': 'VMware ESXi 6.5',
  'provisionable_on': ['c1.small.x86',
                       'baremetal_1',
                       'c1.xlarge.x86',
                       'baremetal_3',
                       'c2.medium.x86',
                       'm1.xlarge.x86',
                       'baremetal_2',
                       'm2.xlarge.x86',
                       's1.large.x86',
                       'baremetal_s',
                       't1.small.x86',
                       'baremetal_0',
                       'x1.small.x86',
                       'baremetal_1e'],
  'slug': 'vmware_esxi_6_5',
  'version': '6.5'},
 {'distro': 'windows',
  'name': 'Windows 2012 R2',
  'provisionable_on': ['c1.small.x86',
                       'baremetal_1',
                       'c1.xlarge.x86',
                       'baremetal_3',
                       'm1.xlarge.x86',
                       'baremetal_2',
                       't1.small.x86',
                       'baremetal_0',
                       'x1.small.x86',
                       'baremetal_1e'],
  'slug': 'windows_2012_r2',
  'version': '2012 R2'},
 {'distro': 'windows',
  'name': 'Windows 2016 Standard',
  'provisionable_on': ['c1.small.x86',
                       'baremetal_1',
                       'c1.xlarge.x86',
                       'baremetal_3',
                       'm1.xlarge.x86',
                       'baremetal_2',
                       't1.small.x86',
                       'baremetal_0',
                       'x1.small.x86',
                       'baremetal_1e'],
  'slug': 'windows_2016',
  'version': '2016'}]
```

**Comment by t0mk** [comment]
I recommend to complain at help@packet.net, to put together a static webpage which would display a table of OS and it's slug, and it's availablity on different plans.

**Comment by jjasghar** [comment]
Should I close this PR then? I was trying to give a future person a quick link to get this information instead of the 20 mins it took me to find it.

**Comment by t0mk** [comment]
Hi @jjasghar thanks for reminding me on this one, and sorry for the delay. It's probably a good idea to mention where to get the slugs, saving time for users. See my one issue in review.

File: website/docs/r/device.html.markdown [hunk 1]
```diff
@@ -119,7 +119,7 @@ The following arguments are supported:
 
 * `hostname` - (Required) The device name
 * `project_id` - (Required) The id of the project in which to create the device
-* `operating_system` - (Required) The operating system slug
+* `operating_system` - (Required) The operating system slug, NOTE: To find the slugs, try clicking on the drop down and converting any spaces or "."s to "_" (underscores). If that doesn't work, visit https://www.packet.net/developers/api/operatingsystems/ and use the API.
```
Comment (t0mk) [inline website/docs/r/device.html.markdown]: Just please mention that the "the drop down" is in the web app, and whereabouts in the web app it is.

**Comment by t0mk** [comment]
@jjasghar I add the doc myself. I only mention the API docs, where you can put the API auth token and see the slugs. I don't mention the web app because
- I think the format of the slugs might be changing, and the slugs themselves are not visible in the web app
- if you use terrafrom, you have the API auth token available somwhere in your env, or in hcl file.

**Pull request merged**